### ISSUE TITLE
bug: Adding surround is not working (gsa" doesn't quotes the current word under cursor)

### ISSUE THREAD

**Issue by bijuli74** [LazyVim/LazyVim#2160]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

Latest release 10.8.2  879e295

### Operating system/version

Windows

### Describe the bug

the docs says to surround the current word under cursor use `gsa` but when using gsa" or gsa( it doesn't do anything

### Steps To Reproduce

move the cursor to the word you want to surround in tags
type gsa"
Eg: Hello -> "Hello" .
But this does not work.

### Expected Behavior

I found out gsaiw" works but then the docs should be update.

### Repro

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  "folke/LazyVim",
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```

**Comment by dpetka2001** [comment]
Did you try reading the docs of the plugin? `gsa` is just a prefix to add surrounding, that accepts as input any valid vim text object motion (it's not just `gsaiw"` like in your case, but can be other things too). I don't believe it has anything to do with LazyVim docs. You can't expect the LazyVim docs to incorporate all the docs of the plugins it uses. Users should be encouraged to also read the docs of the plugins. That's why the LazyVim docs have a link to the github repo of each plugin if you click on the corresponding name of the plugin at the top of each section that describes a plugin.

**Comment by JonatasAmaral** [comment]
Yeah, as stated, selecting what to surround with `iw` is the expected behavior of the plugin.
But i fell i got your point, at first there's no way to know it. Or really **discover** it, as you can in fact know reading docs, but come on, people don't, right?! 😆

I think the WhichKey has exactly this _discoverability_  role, and typing `gsa` could show the "selection" hint (same show for `v`, `y`, etc), to make clear user is in a "select what" state (the cursor does change to `_` in this step, but that's kind of subtle)

And the "setting surrounding" step could also give some WhichKey hints (not just a Noice notification)

But seems like that integration is once again, plugin's responsibility right?!

**Comment by dpetka2001** [comment]
The mode is shown by default in LazyVim (unless you customise your own Lualine and remove it). If you press `gsa` you will see that you're in mode `O-PENDING`. If you read `:h Operator-pending`, you will see that Neovim awaits for a text motion to specify the text that will be operated on. I think that `which-key` has no reason to provide any more visual hints as all information is available to the user. The only thing missing is the user properly reading the docs to know the answer that he needs. I believe that users should be encouraged to read the docs just like a lot of other people have already done and not take it as an offense when being told to actually read the docs (as I have seen in some examples). That's the whole reason the docs exist. To provide the users with answers. I'm sorry that I don't agree with your opinion, that's just what I personally think as a user.

**Comment by JonatasAmaral** [comment]
I _kind of_ don't agree with that "dead the docs" think

Sure that is important, and awesome how it's simplified to get it on vim
Still, it's a thing on good UX to have **discoverability**, where users can find (most of) what they need by minor experimentation at most (a said "case study" of what made iphone so popular back in the day)
And i don't think developer tools should be exception

WhichKey is a spectacular tool in that sense, and is a tool already at hand, that would sure not hurt to use at maximum
As 0-pending is a really less pronounced state change (no color even), i still think that would be a improvement

So i just made that a request on surround.mini if anyone else would find useful:
https://github.com/echasnovski/mini.nvim/issues/620

**Comment by dpetka2001** [comment]
It's perfectly fine to have different opinions as long as each one respects the other. 
I agree with the suggestions you made about the UX experience that could be improved.
Glad you made an issue at `mini.surround`, hope it gets approved by the maintainer of the plugin.

**Comment by echasnovski** [comment]
> But seems like that integration is once again, plugin's responsibility right?!

If "plugin's" here means 'mini.surround', then I don't think so. Its responsibility is to switch to Operator-pending mode after `gsa` (or `sa` with default mappings) and wait for the textobject/motion. As [is showcased in examples inside its help](https://github.com/echasnovski/mini.nvim/blob/333d2d1090c80ac936b960469a6e93982cbaeb21/doc/mini-surround.txt#L73-L75).

**Comment by abeldekat** [comment]
Hi @JonatasAmaral 

> At first there's no way to know it, or really to discover it, as you can in fact know reading docs, but come on, people don't, right?! 

It goes without saying that one can use `LazyVim` without reading any documentation.
However:
When someone creates a **bug** report, one agrees to the following:

```
Before reporting an issue, make sure to read the [documentation](https://github.com/folke/LazyVim) 
and search [existing issues](https://github.com/folke/LazyVim/issues). 

Usage questions such as "How do I...?" 
belong in [Discussions](https://github.com/folke/LazyVim/discussions) and will be closed.

Did you check docs and existing issues?

Make sure you checked all of the below before submitting an issue
I have read all the LazyVim docs *
I have searched the existing issues of LazyVim *
I have searched the existing issues of plugins related to this issue *
```

Some effort and investigation is necessary before creating a bug report in `LazyVim`. That includes obtaining a *basic* understanding* of the plugin at hand and vim in general.

@bijuli74,

> the docs says to surround the current word under cursor use gsa but when using gsa" or gsa( it doesn't do anything

Could you link to that part of the docs? I don't see `gsa` being described as "surround the current word under cursor".

**Comment by abeldekat** [comment]
@bijuli74,

Kindly close this issue.

**Comment by richtong** [comment]
FYI, in the current Lazyvim, it looks like. this is "gz" is the way to get to this not for "gs" which also says "surround", but I'm not sure with all these plugins what I'm accessing

`gzaw"`
